### PR TITLE
Added unit tests for bip39-service

### PR DESCRIPTION
**Pull request by DanielLeushuis**

### CODE REVIEW

File: src/services/bip39-service/bip39-service.spec.ts [hunk 1]
```diff
@@ -182,6 +184,56 @@ describe("BIP39Service", () => {
             }
         );
     });
+
+    it("should return a string array from a string and when already retrieved return it locally", (done) => {
+        spyOn(httpClient, "get").and.returnValue(Observable.of("word1, word2, word3"));
```
Comment (Mathyn) [inline src/services/bip39-service/bip39-service.spec.ts]: The data passed in the promise is not correctly formatted. Instead it should be written as:
```
Observable.of(`
word1
word2
word3
`)
```

This is also the reason the data you are checking on is `['word1, word2, word3']` and not `['word1', 'word2', 'word3']` which it should be.

File: src/services/bip39-service/bip39-service.spec.ts [hunk 2]
```diff
@@ -182,6 +184,56 @@ describe("BIP39Service", () => {
...
+        spyOn(forge.random, "getBytesSync").and.returnValue("a");
+
+        let intarray = new Uint8Array(1);
+        intarray.set([97]);
+
+        expect((<any>service).getRandomBytes(10)).toEqual(intarray);
+    });
+
+    it("should throw because the bytearray is not correct", () => {
+        expect((<any>service).toMnemonic).toThrowError();
```
Comment (Mathyn) [inline src/services/bip39-service/bip39-service.spec.ts:224]: How does this test work? The `toMnemonic` function is not called, or am I reading this wrong?

File: src/services/bip39-service/bip39-service.spec.ts [hunk 3]
```diff
@@ -182,6 +184,56 @@ describe("BIP39Service", () => {
...
+    });
+
+    it("should give back a specfic seed with an empty mnemonic and passphrase", () => {
+        expect(service.toSeed("")).toBe("4ed8d4b17698ddeaa1f1559f152f87b5d472f725ca86d341bd0276f1b61197e21dd5a391f9f5ed7340ff4d4513aab9cce44f9497a5e7ed85fd818876b6eb402e");
+    });
+
+    it("should return 4 spaces of u3000 when 5 words", () => {
```
Comment (Mathyn) [inline src/services/bip39-service/bip39-service.spec.ts]: Import to add in the test name that this only applies to Japanese language

File: src/services/bip39-service/bip39-service.spec.ts [hunk 4]
```diff
@@ -182,6 +184,56 @@ describe("BIP39Service", () => {
...
+        intarray.set([97]);
+
+        expect((<any>service).getRandomBytes(10)).toEqual(intarray);
+    });
+
+    it("should throw because the bytearray is not correct", () => {
+        expect((<any>service).toMnemonic).toThrowError();
+    });
+
+    it("should give back a specfic seed with an empty mnemonic and passphrase", () => {
```
Comment (Mathyn) [inline src/services/bip39-service/bip39-service.spec.ts]: Maybe this was only added for the code coverage but the mnemonic+passphrase to seed is pretty well tested already, right?

**Review by Mathyn — CHANGES_REQUESTED** [review]

File: src/services/bip39-service/bip39-service.spec.ts (hunk 4 above)
Comment (DanielLeushuis) [reply src/services/bip39-service/bip39-service.spec.ts]: Yes true, it was tested with a mnemonic string that had a length, not with an empty mnemonic. This test tests that.

Comment (Mathyn) [reply src/services/bip39-service/bip39-service.spec.ts]: We should take a look at this. Purely from code cleanliness perspective this test should be added to the 'phraseTestVectors' array along with all other tests. However I'm guessing Sonar will not see it as covered this way?

Comment (DanielLeushuis) [reply src/services/bip39-service/bip39-service.spec.ts]: Added default parameter to the toSeed method and removed the test.

**Review by Mathyn — APPROVED** [review]

**Pull request merged**